javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,95 @@
+// Voice assistant
+
+let name = "";
+let todo = [];
+function getReply(command) {
+  let phraseArr = command.toLowerCase().split(" ");
+
+  if (
+    phraseArr[phraseArr.indexOf("is")] &&
+    phraseArr[phraseArr.indexOf("is") + 1] &&
+    !phraseArr.includes("what")
+  ) {
+    name = phraseArr.slice(phraseArr.indexOf("is") + 1, phraseArr.length);
+    console.log("nice to meet you " + name.join(" "));
+  } else if (
+    phraseArr.includes("what") &&
+    phraseArr.includes("name") &&
+    name !== ""
+  ) {
+    console.log("your name is " + name.join(" "));
+  } else if (phraseArr.includes("add") && phraseArr.includes("todo")) {
+    let todoTask = phraseArr
+      .slice(phraseArr.indexOf("add") + 1, phraseArr.indexOf("to"))
+      .join(" ");
+    if (!todoTask) {
+      console.log("please enter some todo !!");
+    } else {
+      todo.push(
+        phraseArr
+          .slice(phraseArr.indexOf("add") + 1, phraseArr.indexOf("to"))
+          .join(" ")
+      );
```

Comment:
`phraseArr
          .slice(phraseArr.indexOf("add") + 1, phraseArr.indexOf("to"))
          .join(" ")` is same as `todoTask`. 
Instead of this you can simply write as todo.push(todoTask).
Try to not repeat the same code. 'DRY' principle is the one of the fundamentals needed for a developer

Reply:
You are write, I will change it